dev/tools/lib/src/update_packages.dart

```diff
@@ -0,0 +1,1493 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
```

Comment:
This command is almost identical to the flutter_tools version, except:

* writes to stderr (warning/error) and print instead of logger APi
* uses Process.run/start and dart:io File instead of package:process/package:file
* uses path instead of filesystem.path